CHANGELOG.md

```diff
@@ -1,3 +1,7 @@
+#### 2.0.5 Maintenance Release
+
+- JSONata leaks internal references in expression when using lambdas (issue https://github.com/jsonata-js/jsonata/issues/691)
```

Comment:
```suggestion
- Fix leaking internal references in expressions when using lambdas (issue #691)
```
Want to make it clear that this is a fix! Also making the issue reference consistent with previous changelog entries.